sda-commons-server-kafka/src/test/java/com/salesforce/kafka/test/listeners/SaslScramListener.java

```diff
@@ -0,0 +1,123 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Based on https://github.com/salesforce/kafka-junit/blob/9675c18b5221a156821d534129b9f03e6b6779b3/kafka-junit-core/src/main/java/com/salesforce/kafka/test/listeners/SaslPlainListener.java
+ */
+package com.salesforce.kafka.test.listeners;
+
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Define and register a SASL_SCRAM listener on a Kafka broker.
+ *
+ * <p>NOTE: Kafka reads in the JAAS file as defined by an Environment variable at JVM start up. This
+ * property can not be set at run time.
+ *
+ * <p>In order to make use of this Listener, you **must** start the JVM with the following:
+ * -Djava.security.auth.login.config=/path/to/your/jaas.conf
+ */
+public class SaslScramListener extends AbstractListener<SaslScramListener> {
+  private static final Logger logger = LoggerFactory.getLogger(SaslPlainListener.class);
+
+  private String username = "";
+  private String password = "";
+
+  /**
+   * Constructor. Only purpose is to emit an ERROR log message if the System environment variable
+   * java.security.auth.login.config has not be set.
+   */
+  public SaslScramListener() {
+    if (!JaasValidationTool.isJaasEnvironmentValueSet()) {
+      logger.error(
+          "Missing required environment variable set: " + JaasValidationTool.JAAS_VARIABLE_NAME);
+    }
+  }
+
+  /**
+   * Setter.
+   *
+   * @param username SASL username to authenticate with.
+   * @return SaslScramListener for method chaining.
+   */
+  public SaslScramListener withUsername(final String username) {
+    this.username = username;
+    return this;
+  }
+
+  /**
+   * Setter.
+   *
+   * @param password SASL password to authenticate with.
+   * @return SaslScramListener for method chaining.
+   */
+  public SaslScramListener withPassword(final String password) {
+    this.password = password;
+    return this;
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  @Override
+  public String getProtocol() {
+    return "SASL_PLAINTEXT";
+  }
+
+  @Override
+  public Properties getBrokerProperties() {
+    final Properties properties = new Properties();
+    properties.put("sasl.enabled.mechanisms", "SCRAM-SHA-256,SCRAM-SHA-512");
+    properties.put("sasl.mechanism.inter.broker.protocol", "SCRAM-SHA-512");
+
+    // properties.put("inter.broker.listener.name", "SASL_PLAINTEXT");
```

Comment:
Can we remove the commented code or add an advice when to use it?

Reply:
No sure what this is for. It came from here and I didn't removed it (https://github.com/salesforce/kafka-junit/blob/9675c18b5221a156821d534129b9f03e6b6779b3/kafka-junit-core/src/main/java/com/salesforce/kafka/test/listeners/SaslPlainListener.java#L90).

I'll remove it.